yarnpkg/yarn
# made integrity hash to be generated based on lockfile entries, not lockfile content

**Summary**
This will fix some of the cases in #569
- Integrity generation in `check` command was based on lockfile on disk content rather than values inside lockfiles.
- Integrity check in `install.js` command can be done after dependencies are resolved, i.e. lock file is read and parsed.

**Test plan**

Added test.

Manual test plan:
```
bestander@DESKTOP-9BB67ID:/mnt/c/Users/besta/work/temp$ ../yarn/bin/yarn install
yarn install v0.17.0-0
warning No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
Done in 0.28s.
bestander@DESKTOP-9BB67ID:/mnt/c/Users/besta/work/temp$ ../yarn/bin/yarn install
yarn install v0.17.0-0
warning No license field
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.12s.
bestander@DESKTOP-9BB67ID:/mnt/c/Users/besta/work/temp$ ../yarn/bin/yarn install --force
yarn install v0.17.0-0
warning No license field
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Rebuilding all packages...
success Saved lockfile.
Done in 0.29s.

bestander@DESKTOP-9BB67ID:/mnt/c/Users/besta/work/temp$ ../yarn/bin/yarn check --integrity
yarn check v0.17.0-0
warning No license field
success Folder in sync.
Done in 0.12s.
bestander@DESKTOP-9BB67ID:/mnt/c/Users/besta/work/temp$ rm node_modules/mime-db/package.json
bestander@DESKTOP-9BB67ID:/mnt/c/Users/besta/work/temp$ ../yarn/bin/yarn check --integrity
yarn check v0.17.0-0
warning No license field
success Folder in sync.
Done in 0.12s.
bestander@DESKTOP-9BB67ID:/mnt/c/Users/besta/work/temp$ ../yarn/bin/yarn check
yarn check v0.17.0-0
warning No license field
error mime-types#mime-db not installed
error Found 1 errors.
info Visit https://yarnpkg.com/en/docs/cli/check for documentation about this command.
```
